addons/account_peppol/models/res_config_settings.py

```diff
@@ -83,9 +83,15 @@ def _call_peppol_proxy(self, endpoint, params=None, edi_user=None):
         return response
 
     # -------------------------------------------------------------------------
-    # COMPUTE METHODS
+    # COMPUTE AND ONCHANGE METHODS
     # -------------------------------------------------------------------------
 
+    @api.onchange('account_peppol_endpoint')
+    def _onchange_account_peppol_endpoint(self):
+        for company in self:
+            endpoint = company.account_peppol_endpoint or ''
+            company.account_peppol_endpoint = ''.join([char for char in endpoint if char.isalnum()])
```

Comment:
`join` accepts an iterator as parameter too
```suggestion
            company.account_peppol_endpoint = ''.join(char for char in endpoint if char.isalnum())
```